### ISSUE TITLE
False positive import error on two files in the same directory with an __init__.py

### ISSUE THREAD

**Issue by DottoreG** [PyCQA/pylint#4226]
### Steps to reproduce

File `m_a.py` imports (empty) `m_b.py`.

### Current behavior

This results in the error message

`m_a.py:1:0: E0401: Unable to import 'm_b' (import-error)`

### Expected behavior

I think the error message is wrong. Python is able to import the module.
Perhaps an underscore in a module is not desired. But then it should be stated as such. (I could not find anything in this regard.)

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.7 (default, Jan 25 2021, 09:13:40) 
[GCC 9.3.0]
```

**Comment by Pierre-Sassoulas** [comment]
I cannot reproduce, am I missing something ? 

$ cat m_a.py
```
import m_b
print(m_b)
```
$ cat m_b.py
```
print("mb")
```

$ pylint m_a.py 
```
------------------------------------
Your code has been rated at 10.00/10
```

**Comment by DottoreG** [comment]
I just used your file contents and ran it again:

$ pylint m_a.py

```
************* Module m_a
m_a.py:1:0: C0114: Missing module docstring (missing-module-docstring)
m_a.py:1:0: E0401: Unable to import 'm_b' (import-error)

-----------------------------------------------------------------------
Your code has been rated at -20.00/10 (previous run: -60.00/10, +40.00)
```
(Did you even not get the message regarding the missing module docstring?)

**Comment by Pierre-Sassoulas** [comment]
``missing-docstring`` was deactivated via pylintrc. Could you upgrade pylint to 2.7.2 first and then could you provide your configuration if the problem still exists ?

**Comment by DottoreG** [comment]
I think I found the missing piece! I performed all my tests in the directory of my current project where I initially observed the issue. I didn't expect to make a difference. But thinking about the problem again I realised that I have an empty `__init__.py` in my directory. And this makes the difference!
So you should be able to reproduce the issue using these three files. Sorry for the confusion.

**Comment by JimiC** [comment]
I stumbled on the same behavior. My use case was that I had `py` files in a child directory without any files in the parent directory:

i.e.
```
dir1
  dir2
     foo.py
```
This was giving false positive `import-error` inside `foo.py` for imported project modules which `Python` was importing without issues.

In my case, the solution was to add an empty `__init__.py` file in the directory `dir1`.

i.e.
```
dir1
  __init__.py
  dir2
     foo.py
```

**Comment by Pierre-Sassoulas** [comment]
I added the investigation label because there are loads of similar issue, maybe linked to namespace handling (?).

**Comment by jacobtylerwalls** [comment]
Closely related to #2862, but might be worth having both test cases, so I think it's okay to keep both open until we know both will be solved by the same PR. Both involve an `__init__.py` somewhere. I'll leave a comment on #2862.

**Comment by jacobtylerwalls** [comment]
But this _is_ the same test case as #3984 with a clean discussion at the top, so closing in favor of that earlier issue.
